develop/devguide/Connector/Troubleshooting/QAction_exceptions/Identifying_the_source_of_an_exception.md

```diff
@@ -0,0 +1,128 @@
+
+However, if you change the code of the getter of the Address to this:
+
+```csharp
+try
+{
+    return RetrieveCustomDomain();
+}
+catch(Exception ex)
+{
+    throw ex;
+}
+```
+
+Then the logging of the exception will result in this:
+
+```csharp
+System.IndexOutOfRangeException: Index was outside the bounds of the array.
+    at MailServer.get_Address()
+    at QAction.Process()
+    at QAction.Run(SLProtocol protocol)
+```
+
+This will be more useful, as you will have more information to go on and you will not keep looking at the `ProcessMethod`. However, it is not ideal yet, as there is no indication that the actual problem is in the `RetrieveCustomDomain` method.
+
+Also, note that adding the "throw ex" will generate two SonarCube warnings:
+
+- S2372: Exceptions should not be thrown from property getters. Property getters should be simple operations that are always safe to call. If exceptions need to be thrown, it’s best to convert the property to a method.
+
+- S3445: Exceptions should not be explicitly rethrown. The reason is because the stack trace is reset and this makes debugging a lot harder, which is ironic here as the original exception was even harder to debug.
+
+### Inline
+
+A problem that could also explain why the stack trace did not show the `get_<PropertyName>` method or the child methods that are called is something known as *Inline*. When the code gets compiled, the compiler will optimize your code and could for example decide that the content of the method will be copied into the actual code that calls the method. This way, the parent method will never call the method but instead immediately execute the code lines of that method. Whenever such lines generate an exception, the stack trace will show the parent method as the location of the issue instead of the child method, as that child method simply does not exist in the compiled code.
+
+Getting a property is actually calling a `get_<PropertyName>` method in the background, which can be optimized away just like any other method. Adding the try-catch in the previous example prevented the optimization from the compiler, which let it show the get method call in the stack trace of the exception.
+
+### Summary
+
+- Like the SonarCube warning mentions, property getters should never cause exceptions. If exceptions are possible, consider catching them and gracefully returning an exception value, or making a method instead of a getter. A property getter is also considered something simple, meaning that the developer calling the property assumes that this can be called thousands of times in a loop without performance impact. If more complexity is behind it, a method is a better choice, as it flags to the developer that it might be better to call the result once, store it in a variable before the loop starts, and then use that variable inside the loop.
```

Comment:
```suggestion
- Like the SonarQube warning mentions, property getters should never cause exceptions. If exceptions are possible, consider catching them and gracefully returning an exception value, or making a method instead of a getter. A property getter is also considered something simple, meaning that the developer calling the property assumes that this can be called thousands of times in a loop without performance impact. If more complexity is behind it, a method is a better choice, as it flags to the developer that it might be better to call the result once, store it in a variable before the loop starts, and then use that variable inside the loop.
```